REPL.cpp

```diff
@@ -131,16 +181,53 @@ bool REPL::ExecuteSwift(std::string line)
                                                                          "swift_repl_module",
                                                                          swift::PrimarySpecificPaths("", input.module_name),
                                                                          m_llvm_ctx));
+
     SetCurrentLoggingArea(LoggingArea::IR);
     if(ShouldLog(LoggingPriority::Info))
     {
+        std::cout << "Symbols in IR:\n";
+        for(auto &g : llvm_module->global_values())
+            std::cout << '\t' << g.getName().str() << '\n';
+
         std::cout << "=========LLVM IR==========\n";
         std::string llvm_ir;
         llvm::raw_string_ostream str_stream(llvm_ir);
         llvm_module->print(str_stream, nullptr);
         str_stream.flush();
         std::cout << llvm_ir << std::endl;
     }
+
+    SetCurrentLoggingArea(LoggingArea::JIT);
+    m_jit->AddModule(std::move(llvm_module));
+
+    swift::Mangle::ASTMangler mangler;
+    std::string mangled_fn_name = mangler.mangleEntity(res_fn, false);
+    std::string mangled_var_name = mangler.mangleGlobalVariableFull(res_var);
+
+    using IntFn = std::add_pointer<uint64_t()>::type;
+    IntFn result_fn = nullptr;
+    uint64_t *result =  nullptr;
+    if(auto symbol = m_jit->LookupSymbol(mangled_fn_name))
+    {
+        result_fn = reinterpret_cast<IntFn>(symbol->getAddress());
+    }
+    else
+    {
+        Log("Failed to load function symbol!", LoggingPriority::Error);
+        return false;
+    }
+    if(auto symbol = m_jit->LookupSymbol(mangled_var_name))
+    {
+        result = reinterpret_cast<uint64_t *>(symbol->getAddress());
+    }
+    else
+    {
+        Log("Failed to load result variable symbol!", LoggingPriority::Error);
+        return false;
+    }
+
+    result_fn();
+    std::cout << (res_var ? *result : 0) << std::endl;
```

Comment:
Log

Reply:
I don't think that this should be a log because this is the actual result of the expression - you shouldn't be able to disable this.